tests/caching/test_integration.py

```diff
@@ -0,0 +1,85 @@
+import pytest
+
+from hamilton import driver
+from hamilton.caching.adapters import SmartCacheAdapter
+from hamilton.caching.store import InMemoryMetadataStore, ShelveResultStore
+
+from tests.caching import cases
+
+
+@pytest.fixture
+def h_driver(request, tmp_path):
+    module = request.param
+    repo = InMemoryMetadataStore(path=tmp_path)
+    cache = ShelveResultStore(path=tmp_path)
+    adapter = SmartCacheAdapter(repo=repo, result_store=cache)
+
+    return driver.Builder().with_modules(module).with_adapters(adapter).build()
+
+
+@pytest.mark.parametrize("h_driver", cases.ALL_MODULES, indirect=True)
+def test_single_run_is_successful(h_driver):
+    """The Driver with the adapter can be executed at least once
+    and return all variables.
+
+    Important to disambiguate more complex bugs
+    """
+    final_vars = []
+    inputs = {}
+
+    for idx, node in enumerate(h_driver.list_available_variables()):
+        if node.name in h_driver.graph.config:
+            continue
+
+        if node.is_external_input:
+            inputs[node.name] = idx
+
+        final_vars.append(node.name)
+
+    h_driver.execute(final_vars, inputs=inputs)
+
+
+@pytest.mark.parametrize("h_driver", cases.ALL_MODULES, indirect=True)
+def test_run_twice_is_successful(h_driver):
```

Comment:
we probably need some more integration tests:

1. Run graph once with input X, then re-run with Input Y. Validate what was computed.
2. Run graph once with input X, change some code. Re-run with input X. Validate what was computed.
3. Run graph once with input X, change some code. Re-run with input Y. Validate what was computed.